cmd/post-install/main.go

```diff
@@ -72,7 +72,9 @@ func isDeploymentReady(logger *zap.SugaredLogger) (bool, error) {
 		totalTime += pollingInterval
 
 		atlasDeployment := mdbv1.AtlasDeployment{}
-		if err := k8sClient.Get(context.TODO(), kube.ObjectKey(namespace, deploymentName), &atlasDeployment); err != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), pollingInterval)
```

Comment:
this sets the timeout to 10 seconds as per https://github.com/mongodb/mongodb-atlas-kubernetes/blob/0d4e7bb6ab54a3aa1781be26113fb4e97c8c941b/cmd/post-install/main.go#L21C2-L22C1 which is too short.

The default request timeout for API server is 60 seconds as per https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/.

Having said that, I wouldn't funnel a timeout here at all. This is the responsibility for the k8sClient. Here, instead I would focus on wiring a proper graceful shutdown context, triggered by SIGTERM and SIGINT, see https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination.

My suggestion here instead is to wire controller-runtime's `SetupSignalHandler` context only, see https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/manager/signals/signal.go#L30 and let the timeouts be handled by the k8s client.

Reply:
Done